arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -787,25 +789,29 @@ Result RocksDBCollection::truncateWithRangeDelete(transaction::Methods& trx) {
   if (!s.ok()) {
     return rocksutils::convertStatus(s);
   }
+  // we need crash server if any failure in order to trigger recovery,
+  // as rocksdb truncate already committed here
```

Comment:
```suggestion
  // We intentionally mark this lambda noexcept, so that when any exception
  // occurs inside the lambda (can happen in case of out of memory for an example),
  // we crash the server instead of continuing with an inconsistent state. 
  // The post-crash recovery will then get everything into a consistent state again.
```

Reply:
:(